crates/wasm-tools-component/src/lib.rs

```diff
@@ -165,9 +169,85 @@ impl exports::Exports for WasmToolsJs {
         core_binary.push(section.id());
         section.encode(&mut core_binary);
 
+        let mut producer_section = wasm_encoder::ProducersSection::new();
+
+        match opts {
+            Some(ref opts) => match &opts.metadata {
+                Some(metadata_fields) => {
+                    for (field_name, items) in metadata_fields {
+                        if field_name != "sdk"
+                            && field_name != "language"
+                            && field_name != "processed-by"
+                        {
+                            return Err(format!("'{field_name}' is not a valid field to embed in the metadata. Must be one of 'language', 'processed-by' or 'sdk'."));
+                        }
+                        let mut field = wasm_encoder::ProducersField::new();
+                        for (name, version) in items {
+                            field.value(&name, &version);
+                        }
+                        producer_section.field(&field_name, &field);
+                    }
+                }
+                None => {}
+            },
+            None => {}
+        };
+
+        core_binary.push(producer_section.id());
+        producer_section.encode(&mut core_binary);
```

Comment:
Instead, please use `add_to_wasm` https://docs.rs/wasm-metadata/latest/wasm_metadata/struct.Producers.html#method.add_to_wasm, which will add to the module's existing producers section of it has one.